WordPress/gutenberg
# MenuGroup: Convert component to TypeScript

## What?
Refactor the `MenuGroup` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactored the component TypeScript
- Refactored tests to TypeScript

## Testing Instructions
1. Run `npm run dev` and ensure no typing errors
2. Run  `npm run test:unit -- packages/components/src/menu-group` and make sure unit tests passes.
3. Test the editor and make sure the MenuGroup still works.
